### PR TITLE
Mmtr 16

### PR DESCRIPTION
**Pull request by thatDevNeal**
- Reduce overall speed.
- Smoother, steady turns.

### CODE REVIEW

File: client/src/js/core/frame.js [hunk 1]
```diff
@@ -16,14 +16,29 @@ Frame.prototype.motor = function(value) {
     // Multiplying by this value should make possible to write directly to PWM
     // Current range is 0 - 127 with first bit as direction
     // Gets 0-100 
-    let k = 1.27;
+    let k = 1.05;
```
Comment (jfeng-dev) [inline client/src/js/core/frame.js]: Does this make a difference? Did you find out `motor()` is called somewhere?

Comment (thatDevNeal) [reply client/src/js/core/frame.js]: it doesn't, but I changed it to stay consistent with `motors()`

File: client/src/js/core/frame.js [hunk 2]
```diff
@@ -16,14 +16,29 @@ Frame.prototype.motor = function(value) {
...
     return Math.round(Math.abs(value * k) | (value & 0x80));
 };
 
-Frame.prototype.motors = function(speed, directions) {   
-    this.motorsArr.forEach ((motor, index) => {
-        this.motorsArr[index] = Math.abs(speed * 1.27) | (directions[index] << 7);
+Frame.prototype.motors = function (speed, directions, directionList) {
+    const stringDirections = directions.toString();
+    const stringDirectionList = [directionList.left.toString(), directionList.right.toString()];
+    const turn = (stringDirectionList.includes(stringDirections));
```
Comment (jfeng-dev) [inline client/src/js/core/frame.js]: unnecessary parenthesis around expression?

File: client/src/js/core/frame.js [hunk 3]
```diff
@@ -16,14 +16,29 @@ Frame.prototype.motor = function(value) {
...
+
+    this.motorsArr.forEach((motor, index) => {
+        if (turn) {
+            if (!directions[index]) {
```
Comment (jfeng-dev) [inline client/src/js/core/frame.js]: Is this `if (!directions[index]){` statement needed? Does this cover the case where the direction from `motors.direction[direction]` doesn't have 4 element in the array? Seems like something we shouldn't worry about.

Comment (thatDevNeal) [reply client/src/js/core/frame.js]: `if (!directions[index])` is true when 0 is the value of the index _when turning_.
It doesn't fall in there when the _index = 0.8_
Which is the wheel that should have a _speed_ of _0_.

File: client/src/js/core/frame.js [hunk 4]
```diff
@@ -16,14 +16,29 @@ Frame.prototype.motor = function(value) {
...
+                return this.motorsArr[index] = Math.abs(speed) | (directions[index] << 7);
+            }
+
+            const slowerIndex = + (index === 2);
```
Comment (jfeng-dev) [inline client/src/js/core/frame.js]: please fix

Comment (thatDevNeal) [reply client/src/js/core/frame.js]: 👍

File: client/src/js/core/frame.js [hunk 5]
```diff
@@ -16,14 +16,29 @@ Frame.prototype.motor = function(value) {
...
+            this.motorsArr[index] = 0 | (directions[index] << 7);
```
Comment (jfeng-dev) [inline client/src/js/core/frame.js]: `directions[index] << 7 || 0`

Comment (thatDevNeal) [reply client/src/js/core/frame.js]: ` || 0` would be useless if written like that, we probably dont need the `0` at all

File: client/src/js/core/frame.js (hunk 1 above)
Comment (jfeng-dev) [reply client/src/js/core/frame.js]: Maybe extract it out into a const and use it in motors as well?

**Review by jfeng-dev — APPROVED** [review]
sure

**Pull request merged**